group-generators/helpers/data-providers/eas/index.ts

```diff
@@ -0,0 +1,123 @@
+import { GraphQLClient } from "graphql-request";
+import { GetAttestationParams, QueryParams } from "./types";
+import { FetchedData } from "topics/group";
+
+export const defaultLimit = 100;
+
+export class EthereumAttestationServiceProvider {
+  public async getAttestationRecipients(
+    params: GetAttestationParams
+  ): Promise<FetchedData> {
+    const dataProfiles: FetchedData = {};
+
+    for await (const item of this._getAttestations(params)) {
+      dataProfiles[item.recipient] = 1;
+    }
+
+    return dataProfiles;
+  }
+
+  public async getAttestationRecipientCount(
+    params: GetAttestationParams
+  ): Promise<number> {
+    const attestations = await this.getAttestationRecipients(params);
+    return Object.keys(attestations).length;
+  }
+
+  public async getAttestationValues(
+    params: GetAttestationParams
+  ): Promise<FetchedData> {
+    const dataProfiles: FetchedData = {};
+
+    for await (const item of this._getAttestations(params)) {
+      const value = isNaN(Number(item.value)) ? 1 : Number(item.value);
+      dataProfiles[item.recipient] = value;
+    }
+
+    return dataProfiles;
+  }
+
+  public async getAttestationValuesCount(
+    params: GetAttestationParams
+  ): Promise<number> {
+    const attestations = await this.getAttestationValues(params);
+    return Object.keys(attestations).length;
+  }
+
+  private async *_getAttestations(params: QueryParams) {
+    const baseUri = `https://${
+      params.network ? `${params.network}.` : ""
+    }easscan.org/graphql`;
+    const client = new GraphQLClient(baseUri);
+    let offset = 0;
+    let hasNext = true;
+
+    while (hasNext) {
+      const res = await client.request(`
```

Comment:
Since it's just 1 query, I kinda like having all the code in the same file. If there will be more, I agree it might make sense to refactor.

But happy to update if you feel it's more aligned..

Reply:
Yes yes you are right, if only 1 query, then we can stay like this with only 1 file 👍